src/app/features/filter-bar/services/state-groups.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Injectable } from '@angular/core';
+import { StateGroup } from '@features/filter-bar/interfaces/state-group';
+
+@Injectable({
+    providedIn: 'root'
+})
+
+export class StateGroupsService {
+
+    stateGroups: StateGroup[] = [
+        {
+            letter: 'A',
+            names: ['Alabama', 'Alaska', 'Arizona', 'Arkansas']
+        },
+        {
+            letter: 'C',
+            names: ['California', 'Colorado', 'Connecticut']
+        },
+        {
+            letter: 'D',
+            names: ['Delaware']
+        },
+        {
+            letter: 'F',
+            names: ['Florida']
+        },
+        {
+            letter: 'G',
+            names: ['Georgia']
+        },
+        {
+            letter: 'H',
+            names: ['Hawaii']
+        },
+        {
+            letter: 'I',
+            names: ['Idaho', 'Illinois', 'Indiana', 'Iowa']
+        },
+        {
+            letter: 'K',
+            names: ['Kansas', 'Kentucky']
+        },
+        {
+            letter: 'L',
+            names: ['Louisiana']
+        },
+        {
+            letter: 'M',
+            names: [
+                'Maine',
+                'Maryland',
+                'Massachusetts',
+                'Michigan',
+                'Minnesota',
+                'Mississippi',
+                'Missouri',
+                'Montana'
```

Comment:
do not hardcode what machine can do
1. create an array of all the states
2. create a method to prepare a group of states by letter
`getStateGroupsByLetter`
3. use this method to get the group in this service